deploy/job-manager/deploy.sh

```diff
@@ -0,0 +1,329 @@
+}
+
+# Tested
+function stderr() {
+    echo -ne ">>| Error Occurred |<< - exiting ...\n"
+    exit 1
+}
+
+# Tested
+function configure_mint_kubernetes() {
+    local ENV=$1
+
+    stdout "Setting to use Google project: project broad-dsde-mint-${ENV}"
+    gcloud config set project broad-dsde-mint-${ENV}
+
+    stdout "Setting to use GKE cluster: project gke_broad-dsde-mint-${ENV}_us-central1-b_listener"
+    kubectl config use-context gke_broad-dsde-mint-${ENV}_us-central1-b_listener
+}
+
+# Tested
+function update_submodule() {
+    local JM_TAG=$1
+    local TOP_LEVEL=$(git rev-parse --show-toplevel)
+
+    stdout "Updating submodule"
+    git submodule update --recursive --remote
+
+    cd "${TOP_LEVEL}/deploy/job-manager/job-manager" && git checkout ${JM_TAG} && cd -
+}
+
+# Tested
+function render_environment_ts() {
+    local CLIENT_ID=""
+
+    stdout "Rendering environment.prod.ts for Job Manager UI"
+    docker run -i --rm \
+        -e CLIENT_ID=${CLIENT_ID} \
+        -v ${PWD}:/working broadinstitute/dsde-toolbox:k8s \
+        /usr/local/bin/render-ctmpl.sh -k /working/environment.template.ts.ctmpl
+}
+
+# Tested
+function inject_angular_and_build_UI() {
+    local ENV=$1
+    local DOCKER_TAG=$2
+    local TOP_LEVEL=$(git rev-parse --show-toplevel)
+
+    stdout "Configuring docker for gcloud"
+    yes | gcloud auth configure-docker
+
+    stdout "Injecting environment.prod.ts into submodule for Job Manager UI"
+    cp "${TOP_LEVEL}/deploy/job-manager/environment.template.ts" "${TOP_LEVEL}/deploy/job-manager/job-manager/ui/src/environments/environment.prod.ts"
+
+    stdout "Building UI docker image: gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-ui:${DOCKER_TAG}"
+    docker build -t gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-ui:${DOCKER_TAG} "${TOP_LEVEL}/deploy/job-manager/job-manager/ui" -f "${TOP_LEVEL}/deploy/job-manager/job-manager/ui/Dockerfile"
+
+    stdout "Pushing UI docker image: gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-ui:${DOCKER_TAG}"
+    docker push gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-ui:${DOCKER_TAG}
+
+    stdout "Resetting the state of submodule after injection"
+    pushd ${TOP_LEVEL}/deploy/job-manager/job-manager
+    git reset --hard HEAD
+    popd
+}
+
+# Tested
+function build_API() {
+    local ENV=$1
+    local DOCKER_TAG=$2
+    local TOP_LEVEL=$(git rev-parse --show-toplevel)
+
+    # Let gcloud to build and push the API container
+    stdout "Building API docker image: gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-api:${DOCKER_TAG}"
+    docker build -t gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-api:${DOCKER_TAG} "${TOP_LEVEL}/deploy/job-manager/job-manager" -f "${TOP_LEVEL}/deploy/job-manager/job-manager/servers/cromwell/Dockerfile"
+
+    stdout "Pushing API docker image: gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-api:${DOCKER_TAG}"
+    docker push gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-api:${DOCKER_TAG}
+}
+
+# Tested
+function create_API_config() {
+    local ENV=$1
+    local config_name=$2
```

Comment:
How about renaming this to `CONFIG_NAME` for consistency with the other variable names?

Reply:
That is better, will make them consistent!